src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -63,11 +72,29 @@ public Assembly LoadFromPath(string fullPath)
             }
         }
 
+        private Assembly TryGetWellKnownAssembly(AssemblyLoadContext context, AssemblyName assemblyName)
+        {
+            if (!_wellKnownAssemblyNames.Contains(assemblyName.Name))
+            {
+                return null;
+            }
+
+            assemblyName.Version = _currentAssemblyVersion;
+
+            return AssemblyLoadContext.Default.LoadFromAssemblyName(assemblyName);
```

Comment:
Is it possible for this call to ever fail? If yes, then mutating the above assemblyName instead of cloning it might cause weird issues, since the mutation flows back into the callers.